src/util/tab-events.js

```diff
@@ -1,57 +1,59 @@
 import eventToPromise from './event-to-promise'
 
+function tabChanged(tabId) {
+    return new Promise((resolve, reject) => {
+        browser.webNavigation.onCommitted.addListener((details)=>{
+            if(details.tabId == tabId && details.frameId == 0)
+                reject();
+        });
+        browser.webNavigation.onHistoryStateUpdated.addListener((details)=>{
+            if(details.tabId == tabId && details.frameId == 0)
+                reject();
+        });
+    });
+}
```

Comment:
I think I made eventToPromise such that it accepts multiple event descriptions to listen for; instead of creating a new Promise each time, could the events defined in `tabChanged` be described in an array that can be added to the `reject` argument of eventToPromise instead?

I am thinking of something like this:
```
const tabChangedEvents = [
    {
        event: browser.tabs.onCommit
        filter: details => (details.tabId == tabId && details.frameId == 0)
    },
    .... same for onHistoryStateUpdated or other events ...
]
```